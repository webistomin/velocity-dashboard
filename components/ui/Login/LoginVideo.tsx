import Vue, { CreateElement, RenderContext, VNode } from 'vue';

export const LoginVideo = Vue.extend({
  name: 'LoginVideo',
  functional: true,
  render(_h: CreateElement, _ctx: RenderContext<{}>): VNode {
    return (
      <div className='login__video-block'>
        <video preload='auto' muted playsInline autoPlay='autoplay' loop='loop' className='login__video'>
          <source src='/video/login-city.webm' type='video/webm; codecs="vp8, vorbis"' />
          <source src='/video/login-city.mp4' type='video/mp4; codecs="avc1.42E01E, mp4a.40.2"' />
          Video tag is not supported in this browser.
        </video>
      </div>
    );
  },
});

export default LoginVideo;
