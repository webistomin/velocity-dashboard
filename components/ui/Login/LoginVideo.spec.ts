import { mount } from '@vue/test-utils';
import { LoginVideo } from './LoginVideo';

describe('LoginVideo', () => {
  const factory = (propsData = {}, options = {}) => {
    return mount(LoginVideo, {
      propsData: {
        ...propsData,
      },
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });
});
