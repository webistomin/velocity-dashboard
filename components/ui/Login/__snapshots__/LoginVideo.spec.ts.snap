// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginVideo Matches snapshot 1`] = `
"<div class=\\"login__video-block\\"><video preload=\\"auto\\" muted=\\"muted\\" playsinline=\\"true\\" autoplay=\\"autoplay\\" loop=\\"loop\\" class=\\"login__video\\">
    <source src=\\"/video/login-city.webm\\" type=\\"video/webm; codecs=&quot;vp8, vorbis&quot;\\">
    <source src=\\"/video/login-city.mp4\\" type=\\"video/mp4; codecs=&quot;avc1.42E01E, mp4a.40.2&quot;\\">Video tag is not supported in this browser.</video></div>"
`;
